app/src/main/java/com/itsaky/androidide/tasks/git/CloneGitTask.java

```diff
@@ -0,0 +1,195 @@
+		textInputEditText = view.findViewById(R.id.textRepo);
+
+		//materialdialog
+		MaterialAlertDialogBuilder materialAlertDialogBuilder = DialogUtils.newMaterialDialogBuilder(activity);
+		materialAlertDialogBuilder.setView(view);
+		materialAlertDialogBuilder.setTitle(R.string.clone_git_repository);
+		materialAlertDialogBuilder.setCancelable(true);
+		materialAlertDialogBuilder.setPositiveButton(R.string.clone, new DialogInterface.OnClickListener() {
+
+			@Override
+			public void onClick(DialogInterface dia, int which) {
+
+				//url
+				String urlRepo = textInputEditText.getText().toString();
+
+				//materialprogress dialog
+				LayoutInflater inflater = (LayoutInflater) activity.getSystemService(Context.LAYOUT_INFLATER_SERVICE);
+				view = inflater.inflate(R.layout.git_dialog_progress, null);
+
+				MaterialAlertDialogBuilder materialAlertDialogBuilder = DialogUtils.newMaterialDialogBuilder(activity);
+				materialAlertDialogBuilder.setTitle(activity.getString(R.string.cloning_repository));
+				materialAlertDialogBuilder.setMessage(urlRepo);
+				materialAlertDialogBuilder.setCancelable(false);
+				materialAlertDialogBuilder.setView(view);
+				
+				materialAlertDialogBuilder.setPositiveButton(R.string.hide, new DialogInterface.OnClickListener() {
+
+					@Override
+					public void onClick(DialogInterface dia, int which) {
+						progressDialog.dismiss();
+					}
+				});
+				materialAlertDialogBuilder.setNegativeButton(R.string.cancel, new DialogInterface.OnClickListener() {
+
+					@Override
+					public void onClick(DialogInterface dia, int which) {
+						mDisposable.dispose();
+					}
+				});
+				progressDialog = materialAlertDialogBuilder.create();
+				progressDialog.show();
+
+				//createObservable
+				createObservable().subscribeOn(Schedulers.io()).observeOn(AndroidSchedulers.mainThread())
+						.subscribe(new Observer<String>() {
+							@Override
+							public void onSubscribe(Disposable disposable) {
+								Log.d(TAG, "onSubscribe: " + disposable);
+                                                                mDisposable = disposable;
+
+							}
+
+							@Override
+							public void onNext(String value) {
+								Log.d(TAG, "onNext: " + value);
+
+							}
+
+							@Override
+							public void onError(Throwable e) {
+								Log.e(TAG, "onError: ", e);
+
+								progressDialog.dismiss();
+								alertDialog.dismiss();
+
+								MaterialAlertDialogBuilder materialAlertDialogBuilder = DialogUtils.newMaterialDialogBuilder(
+										activity);
+								materialAlertDialogBuilder.setTitle(R.string.git_clone_failed);
+
+								//if file exists
+								if (e.toString().contains("org.eclipse.jgit.api.errors.JGitInternalException")) {
+									materialAlertDialogBuilder
+											.setMessage(e.toString().substring(e.toString().lastIndexOf(":") + 2));
+								}
+								//if missing url
+								else if (e.toString().contains("java.net.MalformedURLException: no protocol")) {
```

Comment:
Can we not simply use `e instanceof JGitInternalException/MalformedURLException` or `e.getCause() instanceof JGitInternalException/MalformedURLException`?

Reply:
I tried but i didn't get the correct information can you fix it

Reply:
Okay